cpp/bindings/CMakeLists.txt

```diff
@@ -11,11 +11,21 @@ if(${PYTHON_BINDINGS})
   # Build bindings ###################################################
   find_package(pybind11 REQUIRED)
   message(STATUS "Found ${PYTHON_INCLUDE_DIR} ${PYTHON_EXECUTABLE}")
-  pybind11_add_module(toppra_int toppra_int.cpp bindings.cpp)
+  pybind11_add_module(toppra_int toppra_int.cpp bindings.cpp
+    constraints.cpp paths.cpp)
   target_link_libraries(toppra_int PUBLIC toppra)
   target_include_directories(toppra_int PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
   set_target_properties(toppra_int PROPERTIES
     LIBRARY_OUTPUT_DIRECTORY ${TOPPRA_PYTHON_SOURCE_DIR}
     )
+  if(BUILD_WITH_PINOCCHIO)
+    target_compile_options(toppra_int PRIVATE -DBUILD_WITH_PINOCCHIO)
+    target_link_libraries(toppra_int PUBLIC pinocchio::pinocchio)
+    if(DEFINED PINOCCHIO_WITH_PYTHON_INTERFACE)
+      find_package(Boost REQUIRED COMPONENTS python3)
+      target_compile_options(toppra_int PRIVATE -DPINOCCHIO_WITH_PYTHON_INTERFACE)
+      target_link_libraries(toppra_int PUBLIC Boost::python3)
```

Comment:
This will require Boost right? I am a bit concern with adding boost, since it's a huge library.

Reply:
And some of my packages are running with an outdated version of it (1.5x) and I worry that things will not work properly because of this.

Reply:
Nevermind this comment. Since boost Python is header-only, nothing bad can happen.

Reply:
I understand your concerns. If you reach this point in CMakeLists.txt, it means that pinocchio was installed, with the bindings so boost python is available, with a version compatible with pinocchio. If you don't have boost, you will be restricted to the `jointTorque.Pinocchio(urdfFilename)` API.
So this gives the full bindings to people who have Pinocchio Python bindings, and a restricted API to others.

Reply:
> Nevermind this comment. Since boost Python is header-only, nothing bad can happen.

It isn't header only. There are libraries.